local_planner/include/local_planner/histogram.h

```diff
@@ -55,7 +56,10 @@ class Histogram {
   * @param[in] value, distance to the vehicle of obstacle mapped to (x, y) cell
   *[m]
   **/
-  inline void set_dist(int x, int y, float value) { dist_(x, y) = value; }
+  inline void set_dist(int x, int y, float value) {
+    dist_(x, y) = value;
+    is_empty = false;
```

Comment:
What if a cell is set back to zero? 
We fill all histogram cells with the sum of all distances, then , if not enough points are in one cell it is set back to 0 as we assume it was noise. I think this will not work for this case: https://github.com/PX4/avoidance/blob/978bb05f3fd6b060de4c417809e73318a3e121a4/local_planner/src/nodes/planner_functions.cpp#L134

Reply:
although now don't use the "noise" condition https://github.com/PX4/avoidance/blob/978bb05f3fd6b060de4c417809e73318a3e121a4/local_planner/src/nodes/planner_functions.cpp#L130
In case sometime use it. ...So still need check all histogram cells. I rollback to early stopping.